specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -468,410 +469,422 @@ export function StatsPage(): JSX.Element | null {
     defaultLayout
   );
 
-  return sharedLayout === undefined ? null : (
-    <Form
-      className={`${className.containerFullGray} md:overflow-y-none overflow-y-auto`}
-      onSubmit={(): void => {
-        setState({ type: 'DefaultState' });
-        Promise.all([
-          userPreferences.awaitSynced(),
-          collectionPreferences.awaitSynced(),
-        ]).catch(softFail);
-      }}
-    >
-      <div className="flex flex-wrap items-center gap-2">
-        <H2 className="text-2xl">{statsText.statistics()}</H2>
-        <span className="-ml-2 flex-1" />
-        {pageLastUpdated !== undefined && (
-          <span>
-            {`${statsText.lastRefreshed()} `}
-            <DateElement date={pageLastUpdated} />
-          </span>
-        )}
-        <Button.Gray onClick={(): void => refreshPage()}>
-          {statsText.refresh()}
-        </Button.Gray>
-        {Object.values(layout).every((layouts) => layouts !== undefined) && (
-          <Button.Gray
-            onClick={(): void => {
-              const date = new Date();
-              const sourceIndex = activePage.isShared ? 0 : 1;
-              const pageIndex = activePage.pageIndex;
-              const statsTsv = statsToTsv(
-                layout,
-                activePage.isShared ? 0 : 1,
-                activePage.pageIndex
-              );
-              const sourceName = Object.keys(layout)[sourceIndex];
-              const pageName =
-                Object.values(layout)[sourceIndex]?.[pageIndex].label;
-              const fileName = `Specify 7 Statistics ${sourceName} ${
-                pageName ?? ''
-              } ${date.toDateString()} ${
-                date.toTimeString().split(' ')[0]
-              }.tsv`;
-              downloadFile(fileName, statsTsv).catch(softFail);
-            }}
-          >
-            {statsText.downloadAsTSV()}
-          </Button.Gray>
-        )}
-        {isEditing ? (
-          <>
-            {process.env.NODE_ENV === 'development' && (
+  return (
```

Comment:
(optional) This is not the best way to fix this.

Issues with it:
1. It adds an indent to a lot of code. This is not good because:
   - fewer indent levels left for the rest of the code
   - it destroys the `git blame` history (because every line is technically edited by this change)
2. This is not a full proof fix. While this prevents rendering the elements if there is no permission, it does not prevent running the hooks if there is no permission - maybe it's not a problem right now, but what if that would cause an issue in the future. Or what if some useEffect would expect the content to be rendered, when it's actually not

A better fix for this to to create a component like:
```
export function StatsPage(): JSX.Element {
  return <ProtectedAction ...>
    <ProtectedStatsPage />
  </ProtectedAction;
}
```

Then rename `StatsPage` component you have right now to `ProtectedStatsPage` and remove `export` since it won't need to be exported anymore